src/main/steps/c100-rebuild/address/select/SelectAddressPostController.ts

```diff
@@ -0,0 +1,99 @@
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { C100Address, C100Applicant, C100RebuildPartyDetails, PartyType } from '../../../../app/case/definition';
+import { AppRequest } from '../../../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../../../app/controller/PostController';
+import { Form, FormFields, FormFieldsFn } from '../../../../app/form/Form';
+import { getDataShape, getPartyDetails, updatePartyDetails } from '../../people/util';
+import { C100AddressForm, C100UrlPartyType } from '../definitions';
+import { getUpdatedForm } from '../utils';
+
+import { form as selectAddressForm } from './address-select';
+
+@autobind
+export default class SelectAddressPostController extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    super(fields);
+  }
+
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const { id, partyType } = req.params;
+    const form = new Form(
+      getUpdatedForm(req.session.userCase, id, partyType as C100UrlPartyType, selectAddressForm as C100AddressForm)
+        .fields as FormFields
+    );
+    const { onlycontinue, ...formFields } = req.body;
+    const { _csrf, ...formData } = form.getParsedBody(formFields);
+    const selectedAddressIndex = Number(formData['selectAddress']);
+
+    if (selectedAddressIndex >= 0) {
+      const selectedAddress = req.session.addresses[selectedAddressIndex];
+      req = this.updateAddressDetails(req, id, partyType as C100UrlPartyType, selectedAddress, selectedAddressIndex);
```

Comment:
same here

Reply:
removed the return statement in updateAddressDetails